File: problema 6/c-languaje/problema_6.c
#include <stdio.h>
#include <limits.h>

int main() {
    int matriz[5][5];
    int i, j;
    int     n_menor = INT_MAX, n_mayor = INT_MIN;

    printf("Ingrese los elementos de la matriz:\n");
    for ( i = 0; i < 5; i++){
        for ( j = 0; j < 5; j++){
            scanf("%d", &matriz[i][j]);
        }
    }

    for ( i = 0; i < 5; i++){
        for ( j = 0; j < 5; j++){
            if (matriz[i][j] < n_menor) n_menor = matriz[i][j];
            if (matriz[i][j] > n_mayor) n_mayor = matriz[i][j];
        }    
    }

    printf("El numero menor de la matriz es: %d\n", n_menor);
    printf("El numero mayor de la matriz es: %d\n", n_mayor);

    return 0;
}

File: problema 6/fortran/problema_6.f90
program findMinMax
    implicit none
    integer :: matriz(5,5)
    integer :: i, j
    integer :: n_menor, n_mayor

    ! Initialize variables
    n_menor = huge(n_menor)
    n_mayor = -huge(n_mayor)

    ! Input elements of the matrix
    print *, "Ingrese los elementos de la matriz:"
    do i = 1, 5
        do j = 1, 5
            read *, matriz(i, j)
        end do
    end do

    ! Find the minimum and maximum elements
    do i = 1, 5
        do j = 1, 5
            if (matriz(i, j) < n_menor) then
                n_menor = matriz(i, j)
            endif
            if (matriz(i, j) > n_mayor) then
                n_mayor = matriz(i, j)
            endif
        end do
    end do

    ! Output the results
    print *, "El numero menor de la matriz es: ", n_menor
    print *, "El numero mayor de la matriz es: ", n_mayor

end program findMinMax